packages/flutter/lib/src/material/input_border.dart

```diff
@@ -116,6 +116,99 @@ class _NoInputBorder extends InputBorder {
   }
 }
 
+/// Creates an [InputBorder] that has no painted outline and rounded corners.
+///
+/// The input decorator's "container" is the optionally filled area above the
+/// decorator's helper, error, and counter.
+///
+/// See also:
+///
+///  * [UnderlineInputBorder], the default [InputDecorator] border which
+///    draws a horizontal line at the bottom of the input decorator's container.
+///  * [OutlineInputBorder], an [InputDecorator] border which draws a
+///    rounded rectangle around the input decorator's container.
+///  * [InputDecoration], which is used to configure an [InputDecorator].
+class UnpaintedInputBorder extends InputBorder {
+  /// Creates an unpainted rounded border for an [InputDecorator].
+  ///
+  /// The [borderSide] parameter defaults to [BorderSide.none] (it must not be
+  /// null). Applications typically do not specify a [borderSide] parameter
+  /// because the input decorator substitutes its own, using [copyWith], based
+  /// on the current theme and [InputDecorator.isFocused].
+  ///
+  /// The [borderRadius] parameter defaults to a value where all corners have a
+  /// circular radius of 4.0. The [borderRadius] parameter must not be null.
+  const UnpaintedInputBorder({
```

Comment:
Don't love the name. Makes it sounds like it doesn't do anything at all. I'd say unstroked, nonstroked, strokeless. Get another opinion.